gateway-ha/src/main/java/com/lyft/data/gateway/ha/handler/QueryIdCachingProxyHandler.java

```diff
@@ -111,7 +114,8 @@ public boolean handleAuthRequest(HttpServletRequest request) {
   @Override
   public String rewriteTarget(HttpServletRequest request) {
     /* Here comes the load balancer / gateway */
-    String backendAddress = "http://localhost:" + serverApplicationPort;
+    String backendAddress =
```

Comment:
Can't really tell if it's ok to set the backendAddress to null. Looks like it creates a header later on. Would it be better to not include the header if backendAddress == null? Or is it sufficient as-is?

Reply:
its not really ok, as it will expose a 500 to the client. I should just add a 404 page

Reply:
`null` is a special value for `ProxyServlet.Transparent` that the gateway's `ProxyServeletImpl` extends - if `rewriteTarget` returns a null it will trigger `onRewriteFailure`, which allows sending a 404 back to the client. The problem was that this null value was being intercepted in `ProxyServeletImpl`, which then resulted in a 500. The latest commit fixes that. 

> Looks like it creates a header later on. Would it be better to not include the header if backendAddress == null?
It should return before that due to 
```
if (Strings.isNullOrEmpty(backendAddress)) {
      log.debug("No backend assigned for " + request);
      return null;
    }
```

Reply:
Gotcha. This project is very chatty in general, but I suppose I don't need to add to the noise. Deleted.